app/models/order.rb

```diff
@@ -0,0 +1,21 @@
+class Order < ActiveRecord::Base
+  has_many :order_items
+  has_many :products, through: :order_items
+
+  validates_presence_of :name, :bill_zip, :cc_exp, :cc_num, :city, :email, :sec_code, :state, :street, :zip
+  validates :city, length: { maximum: 30 }
+  # validates :cc_num, length: { is: 4 }
+  validates :status, inclusion: { in: %w(Pending Paid Complete Cancelled) }
```

Comment:
I'm not sure city length is the most important validation, and it would be a good idea to validate credit card number length, zip code length/are numbers, cc expiration date etc.   It was pretty easy to break the app just by typing a long string of numbers into the zip field (it breaks postgres max int size if not a bignum). 
I know validations can be a late priority, but it is good to think about how to best do them at the model level where possible.